src/v/kafka/client/producer.cc

```diff
@@ -64,6 +67,10 @@ make_produce_response(model::partition_id p_id, std::exception_ptr ex) {
 
 ss::future<produce_response::partition>
 producer::produce(model::topic_partition tp, model::record_batch&& batch) {
+    if (_as.abort_requested()) {
+        return ss::make_exception_future<produce_response::partition>(
```

Comment:
I guess this line contradicts with my previous comment. Maybe this should return a failed request instead